dex/router/src/contract.rs

```diff
@@ -83,45 +82,31 @@ pub trait Router:
         mut admins: MultiValueEncoded<ManagedAddress>,
     ) -> ManagedAddress {
         require!(self.is_active(), "Not active");
+
         let owner = self.owner().get();
         let caller = self.blockchain().get_caller();
-
         if caller != owner {
             require!(
                 self.pair_creation_enabled().get(),
                 "Pair creation is disabled"
             );
         }
 
-        require!(first_token_id != second_token_id, "Identical tokens");
```

Comment:
Why did you remove this check?

Reply:
The check is done inside the pair init function. It's duplicated code.